organizing_hub/views/call_views.py

```diff
@@ -292,6 +340,17 @@ def form_valid(self, form):
         """Set local group and owner before save"""
         form.instance.local_group = self.get_local_group()
         form.instance.owner = self.request.user.callprofile
+
+        """Take comma separated caller emails and get or create corresponding
+        call profiles"""
+        caller_emails = form.cleaned_data['caller_emails']
+
+        """Save call_campaign because we need a primary key in order to save
+        Many-to-many caller relationships"""
+        call_campaign = form.save()
```

Comment:
what does `form.save()` do? im familiar with `model.save()` but not `form.save()`. why not call `super(CallCampaignCreateView, self).form_valid(form)` which is how the save was previously being called?

Reply:
```Every ModelForm also has a save() method. This method creates and saves a database object from the data bound to the form.```

https://docs.djangoproject.com/en/dev/topics/forms/modelforms/#the-save-method

We need to do this because before an object is saved we can't access the primary key because it doesn't have one. In order to save many to many relationships you have to have a primary key, or else django will throw an `instance needs to have a primary key value before a many-to-many relationship can be used.` error.

Reply:
https://docs.djangoproject.com/en/1.11/topics/db/examples/many_to_many/

Reply:
Ok I wasn't familiar with this rule.